### PR TITLE
Implement user WebSocket 

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Fixes https://github.com/kubermatic/dashboard/issues/2462. Fixes https://github.com/kubermatic/dashboard/issues/2487.

**Special notes for your reviewer**:

**Release note**:

```release-note
Any changes related to the user and his settings will be immediately visible in the app as user data is downloaded through WebSocket.
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
@lsviben It looks like the new WebSocket is using different data format than the current GET endpoint. Can we change it?

**Comment by maciaszczykm** [comment]
/assign @kgroschoff @floreks 

Can one of you review it?

File: src/app/cluster/cluster-details/version-picker/version-picker.component.ts [hunk 1]
```diff
@@ -52,22 +52,24 @@ export class VersionPickerComponent implements OnInit, OnChanges {
     this.downgradesAvailable = false;
     this.someUpgradesRestrictedByKubeletVersion = false;
 
-    this.upgrades.forEach(upgrade => {
-      const isUpgrade = lt(this.cluster.spec.version, upgrade.version);
-      const isDowngrade = gt(this.cluster.spec.version, upgrade.version);
+    if (this.upgrades) {
```
Comment (floreks) [inline src/app/cluster/cluster-details/version-picker/version-picker.component.ts:55]: This shouldn't be needed. It is always initialized.

Comment (maciaszczykm) [reply src/app/cluster/cluster-details/version-picker/version-picker.component.ts:55]: > It is always initialized.

It isn't. I produces an error during the cluster creation. I can revert this as this is not connected to my change. I just wanted to keep the console clean.

Comment (maciaszczykm) [reply src/app/cluster/cluster-details/version-picker/version-picker.component.ts]: I'll revert since this is probably an issue on the API side.

**Pull request merged**